viewer/packages/api/src/api-helpers/Image360ApiHelper.ts

```diff
@@ -491,15 +492,43 @@ export class Image360ApiHelper<DataSourceT extends DataSourceType> {
     }
   }
 
+  public canDoAction(action: Image360Action): boolean {
+    const insideImage = this._interactionState.currentImage360Entered !== undefined;
+    if (action === Image360Action.Exit) {
+      return insideImage;
+    }
+    if (action === Image360Action.Enter) {
+      if (insideImage) {
+        return false;
+      }
+    } else {
+      if (!insideImage) {
+        return false;
+      }
+    }
```

Comment:
Can you refract `if` `else` to `switch` case, it is more appropriate for `switch` case

Reply:
Done